src/mbgl/tile/geometry_tile_worker.cpp

```diff
@@ -280,7 +282,7 @@ bool GeometryTileWorker::hasPendingSymbolDependencies() const {
 }
 
 void GeometryTileWorker::attemptPlacement() {
-    if (!data || !layers || !placementConfig) {
+    if (!placementConfig) {
```

Comment:
I have a feeling at least one of these guards was needed to ensure that the body of `attemptPlacement` does not run if `setPlacementConfig` is called before `setData` and `setLayers`.